front/lib/api/assistant/conversation.ts

```diff
@@ -200,61 +203,79 @@ async function renderUserMessage(
       ],
     }),
     (async () => {
-      if (userMessage.userId) {
-        return await User.findOne({
-          where: {
-            id: userMessage.userId,
-          },
-        });
+      const userIds = messages
+        .map((m) => m.userMessage?.userId)
+        .filter((id) => !!id) as number[];
+      if (userIds.length === 0) {
+        return [];
       }
-      return null;
+      return await User.findAll({
+        where: {
+          id: userIds,
+        },
+      });
     })(),
   ]);
 
-  return {
-    id: message.id,
-    sId: message.sId,
-    type: "user_message",
-    visibility: message.visibility,
-    version: message.version,
-    created: message.createdAt.getTime(),
-    user: user
-      ? {
-          id: user.id,
-          provider: user.provider,
-          providerId: user.providerId,
-          username: user.username,
-          email: user.email,
-          firstName: user.firstName,
-          lastName: user.lastName,
-          fullName: user.firstName + (user.lastName ? ` ${user.lastName}` : ""),
-          image: null,
-          workspaces: [],
+  return messages.map((message) => {
+    const messageMentions = mentions.filter((m) => m.messageId === message.id);
+    const messageUser =
+      users.find((u) => u.id === message.userMessage?.userId) || null;
+
+    if (!message.userMessage) {
```

Comment:
(We can move this 3 lines above and remove the ? in message.userMessage?)

Reply:
It makes typescript unhappy to remove it 😬

Reply:
Yes didn't mean removing it just moving it in place of line 223 :+1:

Reply:
(to remove the ? at current l.225)

Reply:
done!